### ISSUE TITLE
[BUG] instagrapi.exceptions.BadPassword even with the correct password and login

### ISSUE THREAD

**Issue by RogerMichel** [adw0rd/instagrapi#1498]
**Describe the bug**
`instagrapi.exceptions.BadPassword: The password you entered is incorrect. Please try again.`

**To Reproduce**
```
from instagrapi import Client
import os

from dotenv import load_dotenv
load_dotenv()

def post_to_instagram(username, password, image_path, caption):
    cl = Client()
    cl.login(username, password)


username = os.getenv("INSTAGRAM_USERNAME")
password = os.getenv("INSTAGRAM_PASSWORD")

print(username)
print(password)

image_path = "e97.png"
caption = "I love creating digital art "

post_to_instagram(username, password, image_path, caption)
```

**Traceback**
``` 
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 360, in _send_private_request
    response.raise_for_status()
  File "/Users/rodolphe.berthome/Library/Python/3.8/lib/python/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://i.instagram.com/api/v1/accounts/login/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "upload_to_instagram.py", line 46, in <module>
    post_to_instagram(username, password, image_path, caption)
  File "upload_to_instagram.py", line 10, in post_to_instagram
    cl.login(username, password)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/auth.py", line 432, in login
    logged = self.private_request("accounts/login/", data, login=True)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 533, in private_request
    raise e
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 518, in private_request
    self._send_private_request(endpoint, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 407, in _send_private_request
    raise BadPassword(**last_json)
instagrapi.exceptions.BadPassword: The password you entered is incorrect. Please try again.
```

**Expected behavior**
My password is correct as i just logged on my browser and my mobile with the exact same login.

**Comment by adw0rd** [comment]
I can't reproduce it, figure it out yourself and send us your fixes as PR if the bug really exists

**Comment by R-LP** [comment]
i've tried everything possible, i'm connected on my browser with the same login/password as i use in the script, but on the script i keep getting bad password.

How can i manually export the session information from my browser to dump it to session.json to bypass that ?

**Comment by yueqingliang1** [comment]
Same problem here.

**Comment by yueqingliang1** [comment]
> 

Where is .env?

**Comment by R-LP** [comment]
already tried with hardcoded username and password, got the same error "Bad Password"
im logged in my browser with the exact same credentials (even copy/pasted), is there a way to export the session info from the developper tools of the browser to use in instragrapi ?

**Comment by adw0rd** [comment]
@R-LP you can copy sessionid from cookies and try to log in via `cl.login_by_sessionid(sessionid)` or are you asking something else?

**Comment by adw0rd** [comment]
> Thanks @adw0rd. I can log in by both sessionid and password now. However, it comes to another issue, which is "HTTPError: 401 Client Error: Unauthorized for url:..." as someone raised before. In this case, I can only extract the information of one post then the error occurred.

it at a call of what method? try `cl.inject_sessionid_to_public()`

**Comment by R-LP** [comment]
@adw0rd, i've tried the following:
1. I'm connected on instagram on my browser, on the browser cookies I copy the session_id
2. ```
cl = Client()
session_id = "612123572xxxxxxxxxxxxxxxxxxxxxxxxxxxx"
cl.login_by_sessionid(session_id)
```

but got the following:
`Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 360, in _send_private_request
    response.raise_for_status()
  File "/Users/rodolphe.berthome/Library/Python/3.8/lib/python/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://i.instagram.com/api/v1/feed/timeline/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "upload_to_instagram.py", line 108, in <module>
    print(cl.get_timeline_feed())
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/auth.py", line 241, in get_timeline_feed
    return self.private_request(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 533, in private_request
    raise e
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 518, in private_request
    self._send_private_request(endpoint, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/instagrapi/mixins/private.py", line 386, in _send_private_request
    raise LoginRequired(response=e.response, **last_json)
instagrapi.exceptions.LoginRequired: login_required`


Then when I'm back at my browser I got disconnected from my session, I refresh the page and need to relogin again from the browser.

So it keeps working from the browser but not from the script 🥲

**Comment by Krumil** [comment]
I'm also getting the "instagrapi.exceptions.BadPassword: The password you entered is incorrect. Please try again." error. 
@R-LP Did you find the reason behind this or di you just change the script to use  cl.login_by_sessionid(session_id) method?
Thanks

**Comment by R-LP** [comment]
I haven't find the reason for BadPassword. Also it doesn't work either with the session_id.
So basically so far nothing is working. 🥲

**Comment by falexet** [comment]
Encountered the same issue. Was able to resolve it by changing password of Instagram account.

- Was logged in Safari for couple of weeks
- Tried to execute code to log in using login/password and got 'invalid password'
- Logged out from Instagram in Safari and was able to log in again using the same login/password
- Log in using code still didn't work
- Tried to log in from Chrome browser. Wasn't able to - got the same 'invalid password, please double check' error in UI.
- Logged out from Instagram in Safari and was able to log in again using the same login/password
- Changed account password from Safari in account center to a shorter password (not sure if it matters though)
- Was able to log in into Instagram from both Chrome and code.

Hope this helps.

**Comment by R-LP** [comment]
I've followed the steps of @falexet 
Also realized that when changing the browser (to Chrome), when trying to login with the right password on the UI, got the bad password error.

Changed the password in the account center, managed to connect with both Safari and Chrome.
But still got the bad password error with the script.

Also, here are the logs of the login:

```
https://i.instagram.com/api/v1/launcher/sync/
retro.routes [200] POST https://i.instagram.com/api/v1/launcher/sync/ (269.0.0.18.75, OnePlus 6T Dev)
https://i.instagram.com/api/v1/accounts/login/
retro.routes [400] POST https://i.instagram.com/api/v1/accounts/login/ (269.0.0.18.75, OnePlus 6T Dev)
```

The password is correct for sure, so my guess is that there is a problem with the user-agent or other information related to the login, maybe it's the "OnePlus 6T Dev" device that is blocked?

**Comment by MartianDominic** [comment]
Did all of what you guys recommended. Changed the password, logged out, copied session id, even changed a user agent to another phone model, still nothing works. It log's in, fetches user id from url, fetched media ids and when it goes to like them it logs me out, then gives 400 error, password is incorrect, 403 forbidden for url.

Could it be the cause of update && upgrade I ran today on ubuntu?
When someone solves it, would appreciate you share will all of us, thanks!

**Comment by falexet** [comment]
So this morning the issue is back. Didn't figure out way to resolve it yet.

**Comment by sznyk5252** [comment]
I have exactly the same problem as you, same error with login and password and same error when i try to log in with session id. I try the exact same code on Windows 10 and Linux using google colaboratory on the same browser, and it works on windows but not on Linux

**Comment by adw0rd** [comment]
It is possible that Instagram now blocks accounts like this. Try changing your proxy, you should try

**Comment by MartianDominic** [comment]
Maybe it's updated chromium?Sent from my Galaxy
-------- Original message --------From: sznyk5252 ***@***.***> Date: 01/08/2023  20:11  (GMT+02:00) To: adw0rd/instagrapi ***@***.***> Cc: MartianDominic ***@***.***>, Manual ***@***.***> Subject: Re: [adw0rd/instagrapi] [BUG] instagrapi.exceptions.BadPassword even with the correct password and login (Issue #1498) 
I have exactly the same problem as you, same error with login and password and same error when i try to log in with session id. I try the exact same code on Windows 10 and Linux using google colaboratory on the same browser, and it works on windows but not on Linux

**Comment by ManeFunction** [comment]
It's definitely Instagram fighting bots, got the same error exactly 2 days ago w/o any changes from flawlessly working script for months. Tried everything, but w/o any success. The strange moment is that I can understand if I'm getting this error from my server on Google Clouds, there is a strange device ID and different location, but even if I launch my script locally on my Mac from home – nevermind, the same error. So I'm really curious how they track scripts, but not location and/or device.

**Comment by MartianDominic** [comment]
I am not using proxy, but connecting locallySent from my Galaxy
-------- Original message --------From: Mikhail Andreev ***@***.***> Date: 01/08/2023  21:54  (GMT+02:00) To: adw0rd/instagrapi ***@***.***> Cc: MartianDominic ***@***.***>, Manual ***@***.***> Subject: Re: [adw0rd/instagrapi] [BUG] instagrapi.exceptions.BadPassword even with the correct password and login (Issue #1498) 
It is possible that Instagram now blocks accounts like this. Try changing your proxy, you should try

**Comment by falexet** [comment]
I suspect they might be looking into the headers we provide.

Tried different instagram version/User Agent without success (running from local Mac).

@ManeFunction are you using default UA/Instagram version headers (Instagram 269.0.0.18.75 Android (26/8.0.0; 480dpi; 1080x1920; OnePlus; 6T Dev; devitron; qcom; en_US; 314665256)?

**Comment by MartianDominic** [comment]
I tried using proxies to login through hidemyacc.com. It just blocks instagram whole, every other work, Instagram blocked.Sent from my Galaxy
-------- Original message --------From: Ilia Petrov-Komotskii ***@***.***> Date: 02/08/2023  00:36  (GMT+02:00) To: adw0rd/instagrapi ***@***.***> Cc: MartianDominic ***@***.***>, Manual ***@***.***> Subject: Re: [adw0rd/instagrapi] [BUG] instagrapi.exceptions.BadPassword even with the correct password and login (Issue #1498) 
It's definitely Instagram fighting bots, got the same error exactly 2 days ago w/o any changes from flawlessly working script for months. Tried everything, but w/o any success. The strange moment is that I can understand if I'm getting this error from my server on Google Clouds, there is a strange device ID and different location, but even if I launch my script locally on my Mac from home – nevermind, the same error. So I'm really curious how they track scripts, but not location and/or device.

**Comment by ManeFunction** [comment]
@falexet if this header is within instagrapi, yes, I'm using default. Is there a public method to change it?

**Comment by MartianDominic** [comment]
@ManeFunction I personally have a venv, and went into the auth code part. You may explore that endpoint through github library locally

**Comment by ManeFunction** [comment]
Not sure if I did it right, but I've look at the original code and made my own device info like this:
```
"app_version": "294.0",
"os_version": "16.6",
"dpi": "458dpi",
"resolution": "2688×1242",
"manufacturer": "Apple",
"device": "iPhone12,3",
"model": "iPhone 11 Pro",
"cpu": "A13 Bionic",
"build_id": "20G75",
```
Still the same "wrong password" response :(

**Comment by falexet** [comment]
@ManeFunction these methods allow to get current settings and change them including the UA

cl.get_settings()
cl.set_settings(settings)

As an option user agent can be set directly
cl.set_user_agent('blah blah')


One way to debug this would be to obtain actual headers sent by native Android Instagram app using something like MITM, and then replicate those in the bot. I don't plan to do that for at least couple of days though since I am busy.

**Comment by ManeFunction** [comment]
@falexet I'm using cl.set_device(Dict), is it ok?

**Comment by R-LP** [comment]
Anybody managed to login with instagrapi ?

**Comment by redobsi** [comment]
same issue here

**Comment by thetemplateblog** [comment]
Same issue

**Comment by redobsi** [comment]
Even with a proxy, different proxies, new accounts created on different devices. It ain't working, also it gets instantly the ip of the IP that creates the account blocked & the account gets it's password changed. There are three or four errors that occurs to the same problem. Bad password, login required, wait a moment & try again later.

**Comment by notfounnd** [comment]
The same started here this week. Yesterday I solve my problem using the handler to get 2FA code email. Today login process is failing again.

**Comment by streamcon** [comment]
Same issue

**Comment by sunlightface** [comment]
Continuing Same issue after yesterday

**Comment by MartianDominic** [comment]
@adw0rd Hey Mikhail, I know you're super busy and you said you couldn't replicate the problem. Could you take a look at it again, since so many of us are experiencing it and it's not a one-off issue?
Would be greatly appreciated from the community

**Comment by adw0rd** [comment]
Provide me username, password and proxy to use (I can't reproduce without it)

https://t.me/m11k4

**Comment by falexet** [comment]
@adw0rd people'd be more inclined to help if the request is made politely, consider using words "please/thank you". The issue seems to be expanding and it is in everyone's interests to investigate.

The issue is reproduced on a brand new account without proxy. The location seems to be irrelevant as the issue has been reported from different countries.

**Comment by florifull** [comment]
Same issue here.... started happening yesterday.

It appears there there is a major legitimate issue currently. I am also getting this BadPassword stuff despite using rotating resis... program was working perfectly for last 3 weeks and now getting these issues :(

**Comment by florifull** [comment]
^^I would like to add that this only happened once I tried logging in with a different account that I did not have a session saved for yet.

But now since there is this issue of BadPassword, I cannot get a successful session since I can't log in to begin with..

**Comment by codingthegr8t** [comment]
**exact same issue!**
But I may shed some light on the problem > This from browser console >
Cookie “sessionid” does not have a proper “SameSite” attribute value. Soon, cookies without the “SameSite” attribute or with an invalid value will be treated as “Lax”. This means that the cookie will no longer be sent in third-party contexts. If your application depends on this cookie being available in such contexts, please add the “SameSite=None“ attribute to it. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite 3 checks
Cookie “sessionid” has been rejected for invalid domain. checks
So it is indeed instagram being hard to get

**Comment by wytex** [comment]
Same report here...

`import os
from instagrapi import Client




# Create a new Client object
cl = Client()

# Check if the settings file exists
if os.path.exists('settings.json'):
    # Load the settings from the file
    cl.load_settings('settings.json')
else:
    # Login to Instagram
    cl.login('user', 'psw')

    # Save the settings, including the login session, to a file
    cl.dump_settings('settings.json')

# Get the media ID from the URL
media_id = cl.media_pk_from_url('here_the_url')

# Like the post
cl.media_like(media_id)

print("Liked the post!")`

**Comment by adw0rd** [comment]
> @adw0rd people'd be more inclined to help if the request is made politely, consider using words "please/thank you". The issue seems to be expanding and it is in everyone's interests to investigate.
> 
> The issue is reproduced on a brand new account without proxy. The location seems to be irrelevant as the issue has been reported from different countries.

Please research the problem yourself and send your PR to help others. For all my hundreds and thousands of accounts, this problem is reproduced only if the password is not correct, or if the IP address has been blocked by Instagram and they write as protection that "BadPassword even with the correct password and login"

Thank you very much for your contribution to the research!

**Comment by MartianDominic** [comment]
@adw0rd could the issue be updated chromium to 115?

**Comment by arjahanifard** [comment]
I have the same problem
Sometimes it logs into the accounts and sometimes it doesn't log in at all and it says the password is wrong.

**Comment by niklasdrews** [comment]
Same issue worked for a few months. Restart of the script yield bad password error. Also I can log in normally on the web browser with the same credentials so no IP ban either.

**Comment by florifull** [comment]
> > @adw0rd people'd be more inclined to help if the request is made politely, consider using words "please/thank you". The issue seems to be expanding and it is in everyone's interests to investigate.
> > The issue is reproduced on a brand new account without proxy. The location seems to be irrelevant as the issue has been reported from different countries.
> 
> Please research the problem yourself and send your PR to help others. For all my hundreds and thousands of accounts, this problem is reproduced only if the password is not correct, or if the IP address has been blocked by Instagram and they write as protection that "BadPassword even with the correct password and login"
> 
> Thank you very much for your contribution to the research!

I personally have tried with rotating residential proxies and am also getting this BadPassword error. It worked for weeks and now as of two days ago I am now getting these errors. I don't think its because of any IP address block. 

Thanks for your help.

**Comment by redobsi** [comment]
Hello people. For those with this problem, I still haven't found any fix but I can tell you that it works when you use a brand new account with a good proxy. If you're in a rush like me, look into doing this.

**Comment by darkzbaron** [comment]
same error

**Comment by streamcon** [comment]
I noticed that I can not log into my account through opera and edge browsers. And through chrome I can

**Comment by florifull** [comment]
Was able to login through session from desktop, and even able to scrape things, but once I tried posting I encountered ClipNotUpload and then 403 Client Errors with other long error messages. (Note: I am using location specific rotational residential proxies)

**Comment by florifull** [comment]
@adw0rd was able to login and post with my tester account via resi proxy. I am going to test for myself, it may be because of me not running on Instagrapi latest version, not sure how he was able to do this so easily.

He used my resi proxy to create a client and then logged in  with that proxy.

**Comment by florifull** [comment]
I wasn't able to replicate this myself and got BadPassword again. I hope someone finds a way around this.🙏

**Comment by NicoRulli** [comment]
my only error now is ClipNotUpload. 
It used to work and now it doesn't. 
I don't have a proxy however I never did beforehand and it worked. 

Adw0rd is suggesting I try different clips, however my code does this already as it picks a new media every time it is run and never posts the same clip or pic etc. 

I know instagram is probably cracking down on bots but how can we work around this? 
I've seen talks about headers and 'samesite' attributes. 
Has anyone been able to upload clips lately?

**Comment by antoinep303** [comment]
Hi,
I have the same problem. I'm able to reproduce the issue by opening a session in private mode with Chrome while I log in with the same credentials in a standard session.
I hope it can help.

**Comment by ManeFunction** [comment]
According to the new commit, attached to this issue, I'm almost sure this is not the thing. I was trying a lot of different IP's from different countries and with the script it's always Bad Password, even on my home Mac, which from I successfully logging in using the same creditionals through the browser...

**Comment by ManeFunction** [comment]
Interesting thing, I'm also can't log in even though the browser, if using private mode. My IP is dynamic, so, I assume, it's not IP ban, but _maybe_ our accounts got some kind of shadow ban and allowed only from some kind of trusted devices. For ex. if I initiate a password recovery process and change my password using exact device, I can log in on that device freely after that. But if it's new device, I can't log in even with the right pass. So the main question now, how to initiate a password recovery process and change it from the OS w/o UI shell, is there any existed scripts for this or similar task? Sorry, I'm not a web developer, so all this net communications is kinda hard for me.

**Comment by mohebbi-1386** [comment]
Hello friends

Has your problem been solved by setting a proxy?

**Comment by florifull** [comment]
> Hello friends
> 
> Has your problem been solved by setting a proxy?

No. I am using residential rotating proxies and it doesn't help at all.....

**Comment by MartianDominic** [comment]
I tried using a residential static proxy (in fact I tried a few) to just login to Instagram on my browser and Instagram page doesn't even load :D all others do, not Instagram though :D

**Comment by ManeFunction** [comment]
If my theory is right, IP rotating shouldn't help...

**Comment by antoinep303** [comment]
I found a workaround by enabling 2FA and using pyotp with the generated key from Insta to get the verification code programmatically. 

    import pyotp
    insta_username = Config.INSTAGRAM_USERNAME
    insta_password = Config.INSTAGRAM_PASSWD
    secret= Config.INSTAGRAM_SECRET

    totp = pyotp.TOTP( secret)
    verification_code = totp.now() 

    cl = Client( logger=log)
    cl.login(insta_username, insta_password) # <= return False
    cl.login(insta_username, insta_password, verification_code=verification_code) # <= Return True

**Comment by WhyToFly** [comment]
Having the same issue. I tried the 2FA workaround but now I'm getting a "challenge_required" exception that doesn't trigger either the change_password_handler nor the challenge_code_handler.

**Comment by ManeFunction** [comment]
@antoinep303 tried your solution... it has a potential, but can't manage to work it myself. Every time I'm trying to log in with 2FA enabled through the script, I'm getting "challenge_required" error and when I'm opening IG from app or web next time, it FORCE me to change a password. So, still no luck here :(

**Comment by TheDKey** [comment]
Been having the same problem for more than 2 weeks actually.
Tried using different proxies but no luck.
I'm able to login using browser and app like usual. Just could not go through using script. 

I looked at the login history of my account and found that all Instagrapi login went through a (pseudo?) OnePlus device.
I assume that we all are logging in using this OnePlus device in the perspective of the instagram server, is it correct?
Could this already be flagged?

BTW, instagram seems to block accounts that it considers data-scrapping. Do you guys think making a post once in a while may help? My experimenting instagram account have not made a post but following a lot of other accounts.

**Comment by mateusangelo** [comment]
it was working ob, but im getting this error since yesterday:

The password you entered is incorrect. Please try again.
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://i.instagram.com/api/v1/accounts/login/

During handling of the above exception, another exception occurred:

but the passaword is ok. i can login at chrome and my iphone with the password

**Comment by icpgithub** [comment]
Same problem here. Can log into IG in my browser and same IP (so password is correct and there's no IP ban), but instagrapi is throwing me a badpassword error.

**Comment by im-kuro** [comment]
Yeah same issue, doesn't seem to be a ip or account, I tried logging in with a main account that I've had for a while and it returned `BadPassword`. This is after my code was working fine with no issues for weeks.

**Comment by im-kuro** [comment]
id like to add that i just got it working, all i did was change my password. Anyone else have luck with this?

**Comment by sirengenheiro** [comment]
Currently facing the same issue...

**Comment by lucaspeyrin** [comment]
Do you guys use the "user_agent" parameter with a user_agent pointing to a phone ?

**Comment by yiojo** [comment]
Hello everyone, this worked for me: using a residential Proxy and Instagrapi [Sessions](https://adw0rd.github.io/instagrapi/usage-guide/best-practices.html#:~:text=%2C%203%5D-,Use%20Sessions,-When%20using%20.login) 
Hope it helps:)

**Comment by enakhi** [comment]
If you use 2FA, you will never face this problem.

**Comment by ManeFunction** [comment]
@enakhi you are kinda right, but also wrong... you'll never face this exact problem, but you still won't be able to login because of another issue. Look at my message here 4 days ago.

**Comment by icpgithub** [comment]
I did use 2FA too, but I still got the same problem and after a few tries it forced me to change my password. And it kept falling and asking me again to change the password. So I deactivate 2FA, but still could not log in via instagrapi.

**Comment by johe123qwe** [comment]
Currently facing the same issue...

**Comment by enakhi** [comment]
> @enakhi you are kinda right, but also wrong... you'll never face this exact problem, but you still won't be able to login because of another issue. Look at my message here 4 days ago when 2FA was suggested as a solution for the first time here.

@ManeFunction I have been using 2FA for more than 3 months and never get a BadPassword exception, but 2 weeks ago I tested logging in to an account without enabling 2FA, at first all things were ok, but after 1 week BadPassword appeared, I tried changing password and every solution I think might help, but the only solution that really helped is: enable 2FA for account and login via 2FA.
This solution works for me and maybe for someone else.

**Comment by enakhi** [comment]
> I did use 2FA too, but I still got the same problem and after a few tries it forced me to change my password. And it kept filling and asking me again to change the password. So I deactivate 2FA, but still could not log in via instagrapi.

@icpgithub  So I think using 2FA has solved the BadPAssword problem for you,
If you have a new problem, check why Instagram asks you to change your password.

Sometimes changing your IP helps a lot.

**Comment by icpgithub** [comment]
> > I did use 2FA too, but I still got the same problem and after a few tries it forced me to change my password. And it kept filling and asking me again to change the password. So I deactivate 2FA, but still could not log in via instagrapi.
> 
> @icpgithub So I think using 2FA has solved the BadPAssword problem for you, If you have a new problem, check why Instagram asks you to change your password.
> 
> Sometimes changing your IP helps a lot.

No, it didn't. I could login via 2FA in the browser, with same IP or even changing it just in case. It worked everytime. Except it didn't via instagrapi, when I tried it kept giving me an error when I tried to enter the instagram code and eventually it logged me out and forced me to change the password. So, 2FA works in browser, it doesn't via instagrapi o my end. I did try in two differents computers, differents IP, changing IP blablabla. Same result.

**Comment by ManeFunction** [comment]
@icpgithub exactly the same!

**Comment by darkzbaron** [comment]
2FA made it work for me

**Comment by WhyToFly** [comment]
> Having the same issue. I tried the 2FA workaround but now I'm getting a "challenge_required" exception that doesn't trigger either the change_password_handler nor the challenge_code_handler.

Trying again after a few days (still with 2FA enabled) fixed my issue.

**Comment by mohebbi-1386** [comment]
> با فعال کردن 2FA و استفاده از pyotp با کلید تولید شده از Insta برای دریافت کد بررسی به صورت برنامه ریزی، راه حلی پیدا کردم.
> 
> ```
> import pyotp
> insta_username = Config.INSTAGRAM_USERNAME
> insta_password = Config.INSTAGRAM_PASSWD
> secret= Config.INSTAGRAM_SECRET
> 
> totp = pyotp.TOTP( secret)
> verification_code = totp.now() 
> 
> cl = Client( logger=log)
> cl.login(insta_username, insta_password) # <= return False
> cl.login(insta_username, insta_password, verification_code=verification_code) # <= Return True
> ```

What is the Config code in this piece?

**Comment by akamitoro** [comment]
> cl.login_by_sessionid(sessionid)

Same issue logging in with u/p. I tried logging in with cookie session_id but it won't allow me to upload photo with error: "Could not upload photo: login_required"

Is there any fix for this issue? Thank you

**Comment by akamitoro** [comment]
> I found a workaround by enabling 2FA and using pyotp with the generated key from Insta to get the verification code programmatically.
> 
> ```
> import pyotp
> insta_username = Config.INSTAGRAM_USERNAME
> insta_password = Config.INSTAGRAM_PASSWD
> secret= Config.INSTAGRAM_SECRET
> 
> totp = pyotp.TOTP( secret)
> verification_code = totp.now() 
> 
> cl = Client( logger=log)
> cl.login(insta_username, insta_password) # <= return False
> cl.login(insta_username, insta_password, verification_code=verification_code) # <= Return True
> ```

I keep running into this error?

    raise UnknownError(**last_json)
instagrapi.exceptions.UnknownError: Invalid Parameters

**Comment by arda1984** [comment]
I have the same problem. I tried to use a vpn but no success, I also changed my password couple times via iPhone but didn't work

`instagrapi.exceptions.BadPassword:` The password you entered is incorrect. Please try again. If you are sure that the password is correct, then change your IP address, because it is added to the blacklist of the Instagram Server`

**Comment by chouuul** [comment]
Same issue here ! 
Can't connect with username and password or sessionid, work before.

**Comment by streamcon** [comment]
tried residential soax proxies and other paid services.  when I change the login algorithm to 2FA, the login function works, but I immediately need to change the password on my account

**Comment by icpgithub** [comment]
Still going on after two weeks from not trying to login (and check if there was some hidden time ban by using the script to login).

**Comment by theapk** [comment]
After having this problem for some time, I think it might be resolved for me. The following is what I did to resolve:

I am on a Mac and I am using custom python 3.10 script

1. I logged in to the account I use for the script and enabled 2FA, using `import pyotp` [listed in a post above](https://github.com/subzeroid/instagrapi/issues/1498#issuecomment-1685118747)
2. Log out of any Instagram account/session in my browser, I am using Chrome
3. Got a Proxy from: Webshare.io
4. Added the proxy setting in my code:
`self.cl = Client()`
`proxy = f'https://{proxy_un}:{proxy_pw}@{proxy_url}'`
`self.cl.set_proxy(proxy)`
5. Changed my proxy settings for my Mac OS to use the HTTP & HTTPS with same credentials
6. I added my (HTTP) proxy URL, Username and Password in to System Settings > Proxies
7. (had a few pop ups asking for Keychain Password, so I added the password and then clicked "Always Allow")
8. I then check my Chrome browser to make sure the Public IP had changed with [WhatsMyIP.org](WhatsMyIP.org) and it did.
9. I then logged in to the Instagram account I am trying to use in my script from the browser (Chrome)
10. Then navigated to "Meta Account Center" to change my password, I also added the new password to my script
11. I then grabbed my "sessionid" from "Inspect Console > Application Tab" in my Chrome Browser
12. Added this session ID as a var in my code:
`self.cl = Client()`
`self.cl.login_by_sessionid(<SessionID>)`
14. I then added:
`self.cl.login(os.getenv("UN"), os.getenv("PW"), relogin=False, verification_code=verification_code)`
`self.cl.dump_settings("SessionName.json")`
15. Next I ran my script and connected, also it created my session.json
16. I then commented out my line: `self.cl.login_by_sessionid(<SessionID>)` as I didnt need the session id any longer I had my session.json

Its worked about 4 times now, I will update if it breaks again lol, I hope this will help someone since it was driving me crazy

**Comment by Grypium** [comment]
I had the same issue, logging into the account through the browser and changing the password on the account fixed the API access issue

**Comment by icpgithub** [comment]
> verification_code=verification_code

I've been able to use this method (2FA and pyotp) to login again and upload stories and post via instagrapi. I didn't use proxies. I'll report if this breaks again.

**Comment by spygency1** [comment]
NEED URGENT HELP?!!

NOTE:MESSAGE ONLY THIS ACCOUNT
#🇬🇧Expert Here To Assist In Any Cyber Issues General_Hacking_Recovery_Spy_Blackmail_Stolen,Lost BTC_Hacked Phones ☎️ 📥
🌎Facebook hacking - recovery
🌎Instagram hacking - recovery
🌎Snapchat hacking - recovery
🌎Lost money(investment scams) recovery
🌎Cheating partner spy (online)
🌎Crypto wallet hacking
🌎Phone hacks📱
🌎Message hacks & tweaks

For swift response send a message to my social media accounts, we are active 24/7 to solve all your Online ETHICAL HACKING issues 

INSTAGRAM 📥 spygency.1
EMAIL 📧 spygency@gmail.com

**Comment by spygency1** [comment]
NEED URGENT HELP?!!

NOTE:MESSAGE ONLY THIS ACCOUNT
#🇬🇧Expert Here To Assist In Any Cyber Issues General_Hacking_Recovery_Spy_Blackmail_Stolen,Lost BTC_Hacked Phones ☎️ 📥
🌎Facebook hacking - recovery
🌎Instagram hacking - recovery
🌎Snapchat hacking - recovery
🌎Lost money(investment scams) recovery
🌎Cheating partner spy (online)
🌎Crypto wallet hacking
🌎Phone hacks📱
🌎Message hacks & tweaks

For swift response send a message to my social media accounts, we are active 24/7 to solve all your Online ETHICAL HACKING issues 

INSTAGRAM 📥 spygency.1
EMAIL 📧 spygency@gmail.com

**Comment by dxsteel-labs** [comment]
I know it's a bit old issue but has anyone facing it yet?

I was using the script to just post stories on my profile using my home ip address for a month. But today i got BadPassword exception. I tried to login using 2FA as described here and got challenge_required.
 
Anyone has any solution?

**Comment by rebixpl** [comment]
> I know it's a bit old issue but has anyone facing it yet?
> 
> I was using the script to just post stories on my profile using my home ip address for a month. But today i got BadPassword exception. I tried to login using 2FA as described here and got challenge_required.
> 
> Anyone has any solution?

I don't have solution, but I'm facing the same issue. My code has been working for 3 months posting daily around 60 photos and 60 stories on 20 different accounts. Yesterday I've got my first  BadPassword exception and I can't find a way to fix it.

**Comment by dxsteel-labs** [comment]
> > I know it's a bit old issue but has anyone facing it yet?
> > I was using the script to just post stories on my profile using my home ip address for a month. But today i got BadPassword exception. I tried to login using 2FA as described here and got challenge_required.
> > Anyone has any solution?
> 
> I don't have solution, but I'm facing the same issue. My code has been working for 3 months posting daily around 60 photos and 60 stories on 20 different accounts. Yesterday I've got my first BadPassword exception and I can't find a way to fix it.

Yeah, same. I started facing the issue yesterday after 1 month using the script.

**Comment by rebixpl** [comment]
I think I’ve found  a way to fix it, but I need more testing. If everything works I’ll write how to fix it soon.

**Comment by YomiDr0go** [comment]
> I think I’ve found a way to fix it, but I need more testing. If everything works I’ll write how to fix it soon.
> […](#)

Pls share 🫶🏻

**Comment by rebixpl** [comment]
Yes I fixed it, just looked at my RDP and it’s been working since few hours. When I come back home later today I’ll share everything I’ve done to my code to make it work 😀On Oct 8, 2023, at 20:46, YomiDev ***@***.***> wrote:﻿

I think I’ve found a way to fix it, but I need more testing. If everything works I’ll write how to fix it soon.
…

Pls share 🫶🏻

**Comment by Extrime92** [comment]
> Yes I fixed it,

Man, pls, can you share how you did it? I faced same problem, like everyone here 😀

**Comment by rebixpl** [comment]
> > Yes I fixed it,
> 
> Man, pls, can you share how you did it? I faced same problem, like everyone here 😀

The problem arose when I encountered the BadPassword exception, even though my login credentials were accurate. To resolve this, I thoroughly reworked my login_user function.
1. Below, I'll provide you with the code of my login_user.py file, along with the key improvements I made:
```python 
from instagrapi import Client
from instagrapi.exceptions import LoginRequired
import os


def login_user(username: str, password: str):
    """
    Attempts to log in to Instagram using either the provided session information
    or the provided username and password.
    """

    # login to Instagram
    print("login to Instagram")
    cl = Client()

    # change the proxy
    before_ip = cl._send_public_request("https://api.ipify.org/")
    print(f"Before: {before_ip}")
    print("Changing proxy")
    cl.set_proxy(
        "http://blablabla-zone-resi-region-pl-st-mazovia-city-warsaw:blablabla@pr-eu.pyproxy.com:16666"
    )
    after_ip = cl._send_public_request("https://api.ipify.org/")
    print(f"After: {after_ip}")

    session = None

    if os.path.isfile("./sessions/session_" + username + ".json"):
        session = cl.load_settings("./sessions/session_" + username + ".json")
        print("\n\nDump Settings Already Recorded")

    else:
        cl.dump_settings("./sessions/session_" + username + ".json")
        print("\n\nDump Settings Record Created")

    # client.login(username, password)

    # session = cl.load_settings('./sessions/session_' + username + '.json')

    login_via_session = False
    login_via_pw = False

    if session:
        try:
            cl.set_settings(session)
            cl.login(username, password)

            # check if session is valid
            try:
                cl.get_timeline_feed()
            except LoginRequired:
                print("Session is invalid, need to login via username and password")

                old_session = cl.get_settings()

                # use the same device uuids across logins
                cl.set_settings({})
                cl.set_uuids(old_session["uuids"])

                cl.login(username, password)
            login_via_session = True
            return cl
        except Exception as e:
            print("Couldn't login user using session information: %s" % e)

    if not login_via_session:
        try:
            print(
                "Attempting to login via username and password. username: %s" % username
            )
            if cl.login(username, password):
                login_via_pw = True
                return cl
        except Exception as e:
            print("Couldn't login user using username and password: %s" % e)

    if not login_via_pw and not login_via_session:
        raise Exception("Couldn't login user with either password or session")
```
2. **Proxy Integration:** I introduced a proxy into my script, a cost-effective solution I sourced from PYPROXY.com for just $4 per 1GB. 
<img width="787" alt="image" src="https://github.com/subzeroid/instagrapi/assets/33410494/9ee3a09c-f82d-4101-b9c3-a2efd76ed660">
So change this link for your own:
<img width="738" alt="image" src="https://github.com/subzeroid/instagrapi/assets/33410494/dad88c33-c8a6-4e2c-a41a-a7ba9c864558">
3. **Session Management:** To reduce the need for repetitive logins, I adopted a session management approach. This method entails saving session data to local JSON files, neatly organized in a 'sessions' directory with a filename that includes the username. For a more comprehensive view, you can check the example images below
<img width="579" alt="image" src="https://github.com/subzeroid/instagrapi/assets/33410494/6555d99d-daa9-4aec-b78d-5739eda05705">
With a lot of accounts that would look like this:
<img width="250" alt="image" src="https://github.com/subzeroid/instagrapi/assets/33410494/d7da0f90-4a4f-4ec1-bc1b-80f5427ce206">

4. **Practical Usage Example:** I also provided a concise example of how to employ the login_user function within the context of a project. 
 ```python
                                 # There is at least one image to post
                                  cl = login_user(acc, pw) # log in user with the previous method, it will use session data if it's not
                                                                            # the first time logging in, and save the Client object inside cl variable
  
                                  # get the image from database and save it
                                  url = urls[0]
                                  image = Image.open(requests.get(url, stream=True).raw)
  
                                  image.save(image_path)
  
                                  # handle instagram tasks
                                  handle_instagram_tasks(
                                      cl=cl, # I'm passing cl (Client) object that's been logged in 
                                      igUsername=acc,
                                      account_id=account_id,
                                      url=url,
                                  )
```
5. Function Integration: Within my 'handle_instagram_tasks' function, I seamlessly integrated the Client object obtained from the login_user function. This approach allows for the convenient utilization of Instagrapi's 'post_story' and 'post_photo' methods.
```python
   handle_instagram_tasks(
                                      cl=cl, <----- Client from cl = login_user(acc, pw)
```

**Comment by Akif-Efe** [comment]
> > > Evet düzelttim
> > 
> > 
> > Dostum, nasıl yaptığını paylaşabilir misin lütfen? Buradaki herkes gibi ben de aynı sorunla karşılaştım 😀
> 
> Oturum açma bilgilerim doğru olmasına rağmen sorun BadPassword istisnasıyla karşılaştığımda ortaya çıktı. Bunu çözmek için login_user işlevimi tamamen elden geçirdim.
> 
> 1. Aşağıda size login_user.py dosyamın kodunu ve yaptığım önemli iyileştirmeleri sunacağım:
> 
> ```python
> from instagrapi import Client
> from instagrapi.exceptions import LoginRequired
> import os
> 
> 
> def login_user(username: str, password: str):
>     """
>     Attempts to log in to Instagram using either the provided session information
>     or the provided username and password.
>     """
> 
>     # login to Instagram
>     print("login to Instagram")
>     cl = Client()
> 
>     # change the proxy
>     before_ip = cl._send_public_request("https://api.ipify.org/")
>     print(f"Before: {before_ip}")
>     print("Changing proxy")
>     cl.set_proxy(
>         "http://blablabla-zone-resi-region-pl-st-mazovia-city-warsaw:blablabla@pr-eu.pyproxy.com:16666"
>     )
>     after_ip = cl._send_public_request("https://api.ipify.org/")
>     print(f"After: {after_ip}")
> 
>     session = None
> 
>     if os.path.isfile("./sessions/session_" + username + ".json"):
>         session = cl.load_settings("./sessions/session_" + username + ".json")
>         print("\n\nDump Settings Already Recorded")
> 
>     else:
>         cl.dump_settings("./sessions/session_" + username + ".json")
>         print("\n\nDump Settings Record Created")
> 
>     # client.login(username, password)
> 
>     # session = cl.load_settings('./sessions/session_' + username + '.json')
> 
>     login_via_session = False
>     login_via_pw = False
> 
>     if session:
>         try:
>             cl.set_settings(session)
>             cl.login(username, password)
> 
>             # check if session is valid
>             try:
>                 cl.get_timeline_feed()
>             except LoginRequired:
>                 print("Session is invalid, need to login via username and password")
> 
>                 old_session = cl.get_settings()
> 
>                 # use the same device uuids across logins
>                 cl.set_settings({})
>                 cl.set_uuids(old_session["uuids"])
> 
>                 cl.login(username, password)
>             login_via_session = True
>             return cl
>         except Exception as e:
>             print("Couldn't login user using session information: %s" % e)
> 
>     if not login_via_session:
>         try:
>             print(
>                 "Attempting to login via username and password. username: %s" % username
>             )
>             if cl.login(username, password):
>                 login_via_pw = True
>                 return cl
>         except Exception as e:
>             print("Couldn't login user using username and password: %s" % e)
> 
>     if not login_via_pw and not login_via_session:
>         raise Exception("Couldn't login user with either password or session")
> ```
> 
> 2. **Proxy Entegrasyonu:** Komut dosyama bir proxy ekledim; bu, PYPROXY.com'dan 1 GB başına yalnızca 4 ABD doları karşılığında temin ettiğim uygun maliyetli bir çözümdür.
> 
> <img alt="resim" width="787" src="https://user-images.githubusercontent.com/33410494/275049048-9ee3a09c-f82d-4101-b9c3-a2efd76ed660.png"> Bu bağlantıyı kendinize göre değiştirin: <img alt="resim" width="738" src="https://user-images.githubusercontent.com/33410494/275049154-dad88c33-c8a6-4e2c-a41a-a7ba9c864558.png"> 3. **Oturum Yönetimi:** Tekrarlanan oturum açma ihtiyacını azaltmak için oturum yönetimi yaklaşımını benimsedim. Bu yöntem, oturum verilerinin, kullanıcı adını içeren bir dosya adıyla 'oturumlar' dizininde düzgün bir şekilde organize edilen yerel JSON dosyalarına kaydedilmesini gerektirir. <img alt="resim" width="579" src="https://user-images.githubusercontent.com/33410494/275049697-6555d99d-daa9-4aec-b78d-5739eda05705.png"> Daha kapsamlı bir görünüm için aşağıdaki gibi görünen birçok hesapla aşağıdaki örnek görselleri kontrol edebilirsiniz :<img alt="resim" width="250" src="https://user-images.githubusercontent.com/33410494/275050036-d7da0f90-4a4f-4ec1-bc1b-80f5427ce206.png">
> 4. **Pratik Kullanım Örneği:** Login_user fonksiyonunun bir proje bağlamında nasıl kullanılacağına dair kısa bir örnek de verdim.
> 
> ```python
>                                 # There is at least one image to post
>                                  cl = login_user(acc, pw) # log in user with the previous method, it will use session data if it's not
>                                                                            # the first time logging in, and save the Client object inside cl variable
>  
>                                  # get the image from database and save it
>                                  url = urls[0]
>                                  image = Image.open(requests.get(url, stream=True).raw)
>  
>                                  image.save(image_path)
>  
>                                  # handle instagram tasks
>                                  handle_instagram_tasks(
>                                      cl=cl, # I'm passing cl (Client) object that's been logged in 
>                                      igUsername=acc,
>                                      account_id=account_id,
>                                      url=url,
>                                  )
> ```
> 
> 5. Fonksiyon Entegrasyonu: 'handle_instagram_tasks' fonksiyonumun içerisine, login_user fonksiyonundan elde ettiğim Client nesnesini sorunsuz bir şekilde entegre ettim. Bu yaklaşım, Instagrapi'nin 'post_story' ve 'post_photo' yöntemlerinin rahatlıkla kullanılmasına olanak tanır.
> 
> ```python
>    handle_instagram_tasks(
>                                       cl=cl, <----- Client from cl = login_user(acc, pw)
> ```

Dude, what did you do to fix the erroneous accounts? me "Couldn't login user using username and password: The password you entered is incorrect. Please try again. If you are sure that the password is correct, then change your IP address, because it is added to the blacklist of the Instagram Server " It still gives the error. The codes you gave did not work for me. Do I need to make any edits to the account before running the codes?

**Comment by Akif-Efe** [comment]
> > > Yes I fixed it,
> > 
> > 
> > Man, pls, can you share how you did it? I faced same problem, like everyone here 😀
> 
> The problem arose when I encountered the BadPassword exception, even though my login credentials were accurate. To resolve this, I thoroughly reworked my login_user function.
> 
> 1. Below, I'll provide you with the code of my login_user.py file, along with the key improvements I made:
> 
> ```python
> from instagrapi import Client
> from instagrapi.exceptions import LoginRequired
> import os
> 
> 
> def login_user(username: str, password: str):
>     """
>     Attempts to log in to Instagram using either the provided session information
>     or the provided username and password.
>     """
> 
>     # login to Instagram
>     print("login to Instagram")
>     cl = Client()
> 
>     # change the proxy
>     before_ip = cl._send_public_request("https://api.ipify.org/")
>     print(f"Before: {before_ip}")
>     print("Changing proxy")
>     cl.set_proxy(
>         "http://blablabla-zone-resi-region-pl-st-mazovia-city-warsaw:blablabla@pr-eu.pyproxy.com:16666"
>     )
>     after_ip = cl._send_public_request("https://api.ipify.org/")
>     print(f"After: {after_ip}")
> 
>     session = None
> 
>     if os.path.isfile("./sessions/session_" + username + ".json"):
>         session = cl.load_settings("./sessions/session_" + username + ".json")
>         print("\n\nDump Settings Already Recorded")
> 
>     else:
>         cl.dump_settings("./sessions/session_" + username + ".json")
>         print("\n\nDump Settings Record Created")
> 
>     # client.login(username, password)
> 
>     # session = cl.load_settings('./sessions/session_' + username + '.json')
> 
>     login_via_session = False
>     login_via_pw = False
> 
>     if session:
>         try:
>             cl.set_settings(session)
>             cl.login(username, password)
> 
>             # check if session is valid
>             try:
>                 cl.get_timeline_feed()
>             except LoginRequired:
>                 print("Session is invalid, need to login via username and password")
> 
>                 old_session = cl.get_settings()
> 
>                 # use the same device uuids across logins
>                 cl.set_settings({})
>                 cl.set_uuids(old_session["uuids"])
> 
>                 cl.login(username, password)
>             login_via_session = True
>             return cl
>         except Exception as e:
>             print("Couldn't login user using session information: %s" % e)
> 
>     if not login_via_session:
>         try:
>             print(
>                 "Attempting to login via username and password. username: %s" % username
>             )
>             if cl.login(username, password):
>                 login_via_pw = True
>                 return cl
>         except Exception as e:
>             print("Couldn't login user using username and password: %s" % e)
> 
>     if not login_via_pw and not login_via_session:
>         raise Exception("Couldn't login user with either password or session")
> ```
> 
> 2. **Proxy Integration:** I introduced a proxy into my script, a cost-effective solution I sourced from PYPROXY.com for just $4 per 1GB.
> 
> <img alt="image" width="787" src="https://user-images.githubusercontent.com/33410494/275049048-9ee3a09c-f82d-4101-b9c3-a2efd76ed660.png"> So change this link for your own: <img alt="image" width="738" src="https://user-images.githubusercontent.com/33410494/275049154-dad88c33-c8a6-4e2c-a41a-a7ba9c864558.png"> 3. **Session Management:** To reduce the need for repetitive logins, I adopted a session management approach. This method entails saving session data to local JSON files, neatly organized in a 'sessions' directory with a filename that includes the username. For a more comprehensive view, you can check the example images below <img alt="image" width="579" src="https://user-images.githubusercontent.com/33410494/275049697-6555d99d-daa9-4aec-b78d-5739eda05705.png"> With a lot of accounts that would look like this: <img alt="image" width="250" src="https://user-images.githubusercontent.com/33410494/275050036-d7da0f90-4a4f-4ec1-bc1b-80f5427ce206.png">
> 4. **Practical Usage Example:** I also provided a concise example of how to employ the login_user function within the context of a project.
> 
> ```python
>                                 # There is at least one image to post
>                                  cl = login_user(acc, pw) # log in user with the previous method, it will use session data if it's not
>                                                                            # the first time logging in, and save the Client object inside cl variable
>  
>                                  # get the image from database and save it
>                                  url = urls[0]
>                                  image = Image.open(requests.get(url, stream=True).raw)
>  
>                                  image.save(image_path)
>  
>                                  # handle instagram tasks
>                                  handle_instagram_tasks(
>                                      cl=cl, # I'm passing cl (Client) object that's been logged in 
>                                      igUsername=acc,
>                                      account_id=account_id,
>                                      url=url,
>                                  )
> ```
> 
> 5. Function Integration: Within my 'handle_instagram_tasks' function, I seamlessly integrated the Client object obtained from the login_user function. This approach allows for the convenient utilization of Instagrapi's 'post_story' and 'post_photo' methods.
> 
> ```python
>    handle_instagram_tasks(
>                                       cl=cl, <----- Client from cl = login_user(acc, pw)
> ```

Dude, what did you do to fix the erroneous accounts? me "Couldn't login user using username and password: The password you entered is incorrect. Please try again. If you are sure that the password is correct, then change your IP address, because it is added to the blacklist of the Instagram Server " It still gives the error. The codes you gave did not work for me. Do I need to make any edits to the account before running the codes?

**Comment by shahaksha** [comment]
Any fixes for this? I tried the one suggested by rebixpl but it doesn't work. Also tried most of the other suggestions with multiple accounts.

**Comment by 3az1le** [comment]
The following code might solve your issue.
I had the 'BadPassword' Error then 'TooManyRequest' and 'Chalenge_Required'. The steps are : 

1. Disconnect from your Instagram account
2. Reconnect and get the SESSION_ID from the cookies
3. Put the SESSION_ID in the bot_config file and remove your previous session file if you have one
4. Run the code : it will log without using a proxy the first time so that your session_id isn't flagged ( your request will come from the same IP as the one you used to connect manualy ), then it will dump the session settings to be able to reuse them. Once rerun, it will use the session_file created thanks to the last login.


```python
from instagrapi import Client
import pyotp
import datetime
import os
import json


class InstaUpper:
    def __init__(self, user):
        print(f"Posting on {user}")

        ### Import config 
        with open(f"./configs/{user}_config.json", 'r') as bot_config_file:
            bot_data = json.load(bot_config_file)
        
        with open("./configs/global_config.json", 'r') as global_config_file:
            global_data = json.load(global_config_file)

        USERNAME = bot_data['insta_un']
        INSTA_PSWD = bot_data['insta_psw']
        INSTA_2FA = bot_data['insta_2FA']
        KNOWN_INSTA_SESSION_ID = bot_data['known_insta_session_id']
        PROXY_URL = global_data["PROXY_URL"]
        PROXY_UN = global_data["PROXY_UN"]
        PROXY_PW = global_data["PROXY_PW"] 

        ### Set instance
        self.user = user
        self.SESSION = f"{USERNAME}_session.json"

        secret = INSTA_2FA.replace(" ", "") 
        totp = pyotp.TOTP(secret, interval=30)
        verification_code = totp.now()

        self.cl = Client()

        proxies = f'https://{PROXY_UN}:{PROXY_PW}@{PROXY_URL}'

        before_ip = self.cl._send_public_request("https://api.ipify.org/")
        
        print(f"Before IP: {before_ip}")

        if os.path.isfile(f"./sessions/{user}_session.json"):
            print(f"A previous session for {user} already exist, using proxy")

            self.cl.set_proxy(proxies)
            after_ip = self.cl._send_public_request("https://api.ipify.org/")
            print(f"After: {after_ip}")
            
            self.cl.load_settings(f"./sessions/{user}_session.json")
            assert before_ip != after_ip

            self.cl.login(USERNAME, INSTA_PSWD, relogin=False, verification_code=verification_code)

        else:
            print(f"!! No session for {user} was found, using original IP to sign in with a known SESSION_ID from {user}_config.json")        
            
            self.cl.login_by_sessionid(KNOWN_INSTA_SESSION_ID)

        
        self.cl.dump_settings(f"./sessions/{user}_session.json")`

Hope this works for you guys 😄

**Comment by kiebak3r** [comment]
has anyone managed to get a concrete fix for this? still experiencing this issue. 

Tried to login with session_id also but no such luck

**Comment by subzeroid** [comment]
Closing this as the canonical `BadPassword` support thread rather than a tracked instagrapi code bug.

`BadPassword` can mean the credentials are actually wrong, but Instagram also uses the same error bucket when it does not trust the login context: proxy/IP, device fingerprint, saved session state, account history, or too many fresh password logins. That is why the same password can work in one browser/app/session and fail from a script, or fail only when tested through a particular network identity.

Practical diagnostic order:

1. Upgrade to the current instagrapi release.
2. Try the same username and password in the official Instagram app through the same proxy/IP/network identity used by the script.
3. If the official app also rejects the correct password, treat it as an Instagram-side account/IP/proxy trust state, not an instagrapi parser/client issue.
4. If the app works through the same proxy/IP, inspect script-side device settings, saved session reuse, proxy assignment before `login()`, and login retry flow.

Do not retry `login()` in a loop. Reuse saved settings/sessions, keep one stable clean residential/mobile proxy per account, and use cooldown/manual app login when the account is already flagged.

Documented here for future reports: https://instagrapi.com/guides/errors/bad-password/

Docs backlink commit: https://github.com/subzeroid/instagrapi/commit/2ddeb63c95743b05d6ea3d6a1c67a7a6478d5936

For supported business workflows, prefer the official Instagram Platform API: https://developers.facebook.com/docs/instagram-platform/

If maintaining accounts, sessions, proxies, and challenge handling is the bottleneck, HikerAPI can be a better operational fit: https://hikerapi.com/p/fyfx6ygs
